packages/calling-stateful-client/src/StatefulCallClient.ts

```diff
@@ -206,10 +206,15 @@ export type StatefulCallClientArgs = {
    * you could pass any dummy value like empty string.
    */
   userId: string;
+  /**
+   * Sets the max listeners limit of the 'stateChange' event. Defaults to the node.js EventEmitter.defaultMaxListeners
+   * if not specified.
+   */
+  maxListeners?: number;
```

Comment:
not sure if this should fall under options or args, I can't remember what we were going for in the original distinciton between these two (required vs optional I think), @prprabhu-ms thoughts?

Reply:
This should be under Options.

Args == What should we do.
Options == Modify cross-cutting behaviour related to how we do it.

Reply:
Pasting in some offline discussion:

* Extend CallClientOptions via composition.
* Recommend renaming the field to `maxStateChangeListeners`. The fact that this changes the `EventEmitter` property is an implementation detail. The key idea for user of the API is that this changes how many state change listeners can be added.

So, something like:

```
  export interface StatefulCallClientOptions {
    callClientOptions?: callClientOptions;
    /* Document me */
    maxStateChangeListeners?: number;
  };
```

Reply:
nit: there was a comment one of the previous api suface go-overs to keep types as types instead of interfaces (unless its explicitly an interface a class implements), so this would become:
```javascript
type StatefulCallClientOptions = {
  callClientOptions?: callClientOptions;
  /* Document me */
  maxStateChangeListeners?: number;
}
```

I was thinking we should inline the `callClientOptions`, I don't think users should need to know the difference when putting in options, and instead just have:
```javascript
type StatefulCallClientOptions = callClientOptions & {
  /* Document me */
  maxStateChangeListeners?: number;
}
```

Reply:
Thanks guys for the input. I updated the PR to be:
```
type StatefulCallClientOptions = {
  callClientOptions?: callClientOptions;
  /* Document me */
  maxStateChangeListeners?: number;
}
```
@JamesBurnside I opted to not use the `callClientOptions & {}` option as I like having the two separate. I feel this is more clear which parts of the client your options are affecting. But let me know if you have strong opinion about this or any concerns and I'll change it.

Reply:
Sounds good to me!

Reply:
(This will change the public api format so I will label as a breaking change)